lib/lazy_doc/providers/github_ai.ex

```diff
@@ -0,0 +1,83 @@
+defmodule LazyDoc.Providers.GithubAi do
```

Comment:
if you start implementing multiple providers and they have quite different input/output formats, consider implementing a protocol instead of behaviour.
There is an interesting article that have example on using different providers implementations, using both behaviours and protocols here: https://ulisses.dev/elixir/2021/03/05/aggregation-of-services-pattern.html

Reply:
I was wondering if it was better a protocol or a behavior, the thing is that this function receives a `Req.Response` so every provider when you request data will get a `Req.Reponse` always.

The only thing that changes is the structure of the body.

But since this is handled internally by each provider , I think it can stay as a behavior, unless if you think there could be an exception.

What do you think ?

Reply:
I think in the current state behaviours are good, suggest protocol only if need it in future